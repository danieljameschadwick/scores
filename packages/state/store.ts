import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import themeReducer from "./reducer/ThemeReducer";

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, themeReducer);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [thunk],
});

export const persistor = persistStore(store);

export type AppState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<ReturnType,
  AppState,
  unknown,
  Action<string>>;

export default store;
