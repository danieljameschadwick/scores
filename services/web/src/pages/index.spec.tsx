import React from "react";
import { render, screen, waitFor } from "@testing-library/react-native";
import { Provider } from "react-redux";
import store, { persistor } from "@scores/state/store";
import { PersistGate } from "redux-persist/integration/react";
import Index from "./index";

const customRender = () => {
  return render(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Index />
      </PersistGate>
    </Provider>
  );
};

describe("Index page", () => {
  it("renders base Index", async () => {
    const { queryByTestId } = customRender();

    const carouselScrollWrapper = queryByTestId("carousel-scroll-wrapper");

    // await waitFor(() => expect(carouselScrollWrapper).toBeTruthy());

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("Index renders expected panels", async () => {
    const { getByText } = customRender();

    waitFor(() => expect(getByText("Football Panel")).toBeInTheDocument());
    // await waitFor(() => expect(queryByTestId("carousel-scroll-wrapper")).toBeTruthy());
    // await waitFor(() => expect(queryByTestId("football-panel")).toBeTruthy());
    // await waitFor(() => expect(queryByTestId("cricket-panel")).toBeTruthy());

    // console.log(queryByTestId("football-panel"));

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
