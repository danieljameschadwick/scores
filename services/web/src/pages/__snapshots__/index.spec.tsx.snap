// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page Index renders expected panels 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "minHeight": "100%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexDirection": "column",
        "alignItems": "center",
        "backgroundColor": "rgba(248,248,248,1.00)",
        "borderBottomColor": "rgba(215,220,224,1.00)",
        "borderLeftColor": "rgba(215,220,224,1.00)",
        "borderRightColor": "rgba(215,220,224,1.00)",
        "borderTopColor": "rgba(215,220,224,1.00)",
        "display": "flex",
        "flexDirection": "column",
        "msFlexAlign": "center",
        "msFlexDirection": "column",
        "position": "fixed",
        "width": "100%",
        "zIndex": 10,
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "backgroundColor": "rgba(248,248,248,1.00)",
          "borderBottomColor": "rgba(215,220,224,1.00)",
          "borderBottomWidth": "1px",
          "borderLeftColor": "rgba(215,220,224,1.00)",
          "borderRightColor": "rgba(215,220,224,1.00)",
          "borderTopColor": "rgba(215,220,224,1.00)",
          "width": "100%",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitFlexDirection": "row",
            "display": "flex",
            "flexDirection": "row",
            "height": "50px",
            "msFlexDirection": "row",
            "paddingBottom": "12px",
            "paddingLeft": "12px",
            "paddingRight": "12px",
            "paddingTop": "12px",
            "width": "100%",
          }
        }
      >
        <div
          className="css-cursor-18t94o4 css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          role="link"
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "WebkitAlignSelf": "center",
                "alignSelf": "center",
                "color": "rgba(255,113,0,1.00)",
                "fontSize": "17px",
                "fontWeight": "bold",
                "msFlexItemAlign": "center",
                "msGridRowAlign": "center",
              }
            }
          >
            scores
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitAlignItems": "flex-end",
              "WebkitBoxAlign": "end",
              "WebkitBoxFlex": 1,
              "WebkitBoxPack": "center",
              "WebkitFlexGrow": 1,
              "WebkitJustifyContent": "center",
              "alignItems": "flex-end",
              "flexGrow": 1,
              "justifyContent": "center",
              "marginRight": "10px",
              "msFlexAlign": "end",
              "msFlexPack": "center",
              "msFlexPositive": 1,
            }
          }
        >
          <div
            className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
            onClick={[Function]}
            onContextMenu={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "WebkitTransitionDuration": "0s",
                "transitionDuration": "0s",
              }
            }
            tabIndex="0"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï•Š
            </Text>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitAlignSelf": "center",
              "WebkitBoxDirection": "normal",
              "WebkitBoxOrient": "horizontal",
              "WebkitFlexDirection": "row",
              "alignSelf": "center",
              "display": "flex",
              "flexDirection": "row",
              "msFlexDirection": "row",
              "msFlexItemAlign": "center",
              "msGridRowAlign": "center",
            }
          }
        >
          <div
            className="css-cursor-18t94o4 css-text-901oao r-cursor-1loqt21"
            dir="auto"
            onClick={[Function]}
            role="link"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "paddingBottom": "6px",
                "paddingLeft": "6px",
                "paddingRight": "6px",
                "paddingTop": "6px",
              }
            }
            tabIndex="0"
          >
            Login
          </div>
          <div
            className="css-cursor-18t94o4 css-text-901oao r-cursor-1loqt21"
            dir="auto"
            onClick={[Function]}
            role="link"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "paddingBottom": "6px",
                "paddingLeft": "6px",
                "paddingRight": "6px",
                "paddingTop": "6px",
              }
            }
            tabIndex="0"
          >
            Register
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "backgroundColor": "rgba(248,248,248,1.00)",
          "borderBottomColor": "rgba(215,220,224,1.00)",
          "borderBottomWidth": "1px",
          "borderLeftColor": "rgba(215,220,224,1.00)",
          "borderRightColor": "rgba(215,220,224,1.00)",
          "borderTopColor": "rgba(215,220,224,1.00)",
          "width": "100%",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitFlexDirection": "row",
            "display": "flex",
            "flexDirection": "row",
            "msFlexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          dataSet={
            Object {
              "media": undefined,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(237, 238, 240)",
                "borderLeftWidth": 1,
                "display": "flex",
                "height": 60.5,
                "position": "relative",
                "width": "100%",
                "zIndex": 10,
              },
              Object {
                "backgroundColor": "rgb(248, 248, 248)",
                "borderColor": "rgb(215, 220, 224)",
              },
            ]
          }
        >
          <View
            dataSet={
              Object {
                "media": undefined,
              }
            }
            style={
              Array [
                Object {
                  "borderRightWidth": 1,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": "100%",
                  "overflow": "hidden",
                  "width": "100%",
                },
                Object {
                  "backgroundColor": "rgb(248, 248, 248)",
                  "borderColor": "rgb(215, 220, 224)",
                },
              ]
            }
          >
            <View
              dataSet={
                Object {
                  "media": undefined,
                }
              }
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRightWidth": 1,
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 160,
                  },
                  Object {
                    "backgroundColor": "rgb(248, 248, 248)",
                    "borderColor": "rgb(215, 220, 224)",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "position": "relative",
                      "zIndex": 20,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgb(232, 234, 237)",
                      "borderColor": "rgb(215, 220, 224)",
                      "borderRadius": 5,
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "minWidth": 130,
                      "opacity": 1,
                      "paddingLeft": 15,
                      "paddingRight": 10,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "600",
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000",
                        },
                      ]
                    }
                  >
                    August
                  </Text>
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Entypo",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï…„
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "display": "flex",
                          "flexDirection": "column",
                          "position": "absolute",
                          "top": 45,
                          "zIndex": 10,
                        },
                        Object {
                          "display": "none",
                        },
                      ],
                      Object {
                        "backgroundColor": "rgb(232, 234, 237)",
                        "borderColor": "rgb(215, 220, 224)",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderRadius": 5,
                          "paddingHorizontal": 5,
                          "paddingVertical": 5,
                          "position": "fixed",
                        },
                        Object {
                          "backgroundColor": "rgb(232, 234, 237)",
                          "borderColor": "rgb(215, 220, 224)",
                        },
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "paddingHorizontal": 10,
                              "paddingVertical": 5,
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                              },
                            ]
                          }
                        >
                          August
                        </Text>
                      </View>
                    </View>
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "paddingHorizontal": 10,
                              "paddingVertical": 5,
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                              },
                            ]
                          }
                        >
                          September
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              dataSet={
                Object {
                  "media": undefined,
                }
              }
              style={
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgb(232 234 237)",
                      "borderRightWidth": 1,
                      "display": "flex",
                      "justifyContent": "center",
                      "minWidth": 55,
                      "paddingHorizontal": 20,
                    },
                    Object {},
                  ],
                  Object {
                    "backgroundColor": "rgb(248, 248, 248)",
                    "borderColor": "rgb(215, 220, 224)",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 11,
                      "fontWeight": "600",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#000",
                    },
                  ]
                }
              >
                Football
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "gap": 6,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <ActivityIndicator
                color="#000"
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                    },
                  ]
                }
              >
                Loading...
              </Text>
            </View>
          </View>
        </View>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "backgroundColor": "rgba(249,249,249,1.00)",
        "height": "100%",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "107.5px",
        "paddingBottom": "50px",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "15px",
          "width": "660px",
        }
      }
    >
      <View
        style={
          Array [
            Array [
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderRadius": 10,
                "marginBottom": 10,
                "paddingBottom": 16,
                "paddingHorizontal": 16,
                "paddingTop": 8,
              },
            ],
            Object {
              "backgroundColor": "rgb(232, 234, 237)",
              "borderColor": "rgb(215, 220, 224)",
            },
          ]
        }
        testID="football-panel"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "#dcdddf",
                  "borderBottomWidth": 1,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginBottom": 10,
                  "paddingBottom": 5,
                },
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginRight": 3,
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï‹²
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 13,
                  "fontWeight": "800",
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Football
          </Text>
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Array [
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderRadius": 10,
                "marginBottom": 10,
                "paddingBottom": 16,
                "paddingHorizontal": 16,
                "paddingTop": 8,
              },
            ],
            Object {
              "backgroundColor": "rgb(232, 234, 237)",
              "borderColor": "rgb(215, 220, 224)",
            },
          ]
        }
        testID="cricket-panel"
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "#dcdddf",
                  "borderBottomWidth": 1,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginBottom": 10,
                  "paddingBottom": 5,
                },
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginRight": 3,
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ó°µ­
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 13,
                  "fontWeight": "800",
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Cricket
          </Text>
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "padding": 15,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              No games available.
            </Text>
          </View>
        </View>
      </View>
    </div>
  </div>
</div>
`;

exports[`Index page renders base Index 1`] = `null`;
